gimmicks/Gimmicks.py

```diff
@@ -60,6 +60,10 @@ def __init__(self, fobj):
             self.param_names = ('param0', 'param1', 'param2', 'param3',
                                 'param4', 'param5')
 
+        for i, name in enumerate(self.param_names):
+            if name[0:5] != "param":
+                setattr(type(self), name, Param_Descriptor("param" + str(i)))
```

Comment:
should this just be `setattr(self, name, etc)` not `type(self)`? Otherwise aren't you setting it on an inuninstantiated version of the class?

Reply:
Typically you set descriptors for the class itself instead of an instance. Just like the getters and setters are. I am forgetting every word to explain it right now, but this is how I intended it. My history isn't showing me the sites I got the info from either.